Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -174,15 +51,19 @@ void LoadSettings()
 	GetUserDataPath(datapath);
 
 	// Create the directory if it doesn't exist
-	string moddir = string(datapath) + R"(\Accts\MultiPlayer\docking_module\)";
+	string moddir = string(datapath) + "\\Accts\\MultiPlayer\\docking_module\\";
 	CreateDirectoryA(moddir.c_str(), 0);
 
+	string scCarrierDataFile = moddir + "mobile_docking.ini";
+
 	// Plugin configuration
 	char szCurDir[MAX_PATH];
 	GetCurrentDirectory(sizeof(szCurDir), szCurDir);
 	string scPluginCfgFile = string(szCurDir) + "\\flhook_plugins\\dockingmodules.cfg";
 
 	int dockingModAmount = 0;
+	int carrierCount = 0;
+	int dockedCount = 0;
 	INI_Reader ini;
 	if (ini.open(scPluginCfgFile.c_str(), false))

```

Comment:
Looking at all the ini reading below, you need to add more safety checks that everything is in an intended format. If any file is malformed or has issues you will face issues like you had from the PoB issues the other day.

Reply:
I will also reiterate when you are saving files you should do so to a temp file first as a buffer, then copy the contents. This is protection against corruption. Additionally, saving to separate files to prevent everything dying if one goes blep